modules/primaries/main.tf

```diff
@@ -56,8 +62,167 @@ resource "google_compute_instance_group" "main" {
     name = "install-dashboard"
     port = var.vpc_install_dashboard_tcp_port
   }
-  named_port {
-    name = "kubernetes"
+}
+
+resource "google_compute_health_check" "kubernetes_api" {
+  provider = google-beta
+
+  name = "${local.prefix}k8s"
+
+  check_interval_sec = 10
+  description        = "The Kubernetes API on the TFE primaries."
+  # Beta
+  log_config {
+    enable = true
+  }
+  tcp_health_check {
+    port = var.vpc_kubernetes_tcp_port
+  }
+  unhealthy_threshold = local.instance_count
+}
+
+resource "google_compute_region_backend_service" "main" {
+  health_checks = [google_compute_health_check.kubernetes_api.self_link]
+  name          = local.name
+
+  dynamic "backend" {
+    for_each = google_compute_instance_group.main[*].self_link
+    content {
+      group = backend.value
+
+      description = "One of the TFE primaries."
+    }
+  }
+  description = "The TFE primaries."
+  protocol    = "TCP"
+  timeout_sec = 10
+}
+
+resource "google_compute_address" "main" {
+  name = local.name
+
+  address_type = "INTERNAL"
+  description  = "The private IP address for the TFE primaries."
+  subnetwork   = var.vpc_subnetwork_self_link
+}
+
+resource "google_compute_forwarding_rule" "main" {
+  name = local.name
+
+  backend_service       = google_compute_region_backend_service.main.self_link
+  description           = "Forward to the TFE primaries traffic received by the private IP address."
+  ip_address            = google_compute_address.main.address
+  ip_protocol           = "TCP"
+  load_balancing_scheme = "INTERNAL"
+  network               = var.vpc_network_self_link
+  ports                 = local.ports
+  subnetwork            = var.vpc_subnetwork_self_link
+}
+
+resource "google_compute_instance_template" "load_balancer" {
+  disk {
+    source_image = "ubuntu-1804-lts"
```

Comment:
I'm assuming that for this, if a customer is requiring a specific image here, that they will need to configure this module separately?

Reply:
This hard-coding is a carried over from the original ILB module. At the moment, we have an APT-specific script to setup the internal primaries load balancer nodes so we could make this variable with the caveat that the selected image must include APT.

Reply:
Fixed by https://github.com/hashicorp/terraform-google-terraform-enterprise/pull/73/commits/12876f05efbb679f278ad1f50d9e6812816c906f.